lib/irb.rb

```diff
@@ -979,13 +984,21 @@ def run(conf = IRB.conf)
       end
 
       begin
-        catch(:IRB_EXIT) do
+        @forced_exit = false
```

Comment:
Why does it have to be an ivar instead of a local?

Reply:
not need to. I just updated this part.